xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp

```diff
@@ -150,21 +150,19 @@ enum AVPixelFormat CDVDVideoCodecFFmpeg::GetFormat( struct AVCodecContext * avct
   const char* pixFmtName = av_get_pix_fmt_name(*fmt);
 
   // if frame threading is enabled hw accel is not allowed
-  if(ctx->m_decoderState != STATE_HW_SINGLE)
+  // 2nd condition:
```

Comment:
this block had a return statement and would have required to set deint methods before return.